inst/include/dust/types.hpp

```diff
@@ -291,6 +295,59 @@ T* shared_copy(T* dest, const T src) {
   return dest + 1;
 }
 
+class openmp_errors {
+public:
+  openmp_errors() : openmp_errors(0) {
+  }
+  openmp_errors(size_t len) :
+    count(0), err(len), seen(len) {
+  }
+
+  void reset() {
+    count = 0;
+    std::fill(seen.begin(), seen.end(), false);
+    std::fill(err.begin(), err.end(), "");
+  }
+
+  bool unresolved() const {
+    return count > 0;
+  }
+
+  template <typename T>
+  void capture(const T& e, size_t i) {
+    err[i] = e.what();
+    seen[i] = true;
+  }
+
+  void report(size_t n_max = 4) {
```

Comment:
Note that only the top four are being reported in the message?

Reply:
that is reported at the moment:

```
Error: 10 particles reported errors.
  - 1: Invalid call to rbinom with n = -10, p = 0.0237906
  - 2: Invalid call to rbinom with n = -10, p = 0.0237906
  - 3: Invalid call to rbinom with n = -10, p = 0.0237906
  - 4: Invalid call to rbinom with n = -10, p = 0.0237906
  - (and 6 more)
```

if you have fewer than 4 it looks like:

```
Error: 2 particles reported errors.
  - 4: Invalid call to rbinom with n = -1, p = 0.0237906
  - 9: Invalid call to rbinom with n = -10, p = 0.0237906
```

Reply:
I wondered about making this configurable but honestly I think that's more drama than it's worth. The most likely situation seems to be we get one or two (which this will cope with nicely and give a reasonable length error) or they *all* die in which case we don't want to have a billion lines printed. Probably on error the user will have to get involved with gdb to really progress anyway